### PR TITLE
enable specific ranges

### LINKED ISSUE #123: add in --start and --end kwargs for running mdet

### PR DESCRIPTION
**Pull request by beckermr**
This PR closes #123

### CODE REVIEW

File: pizza_cutter/metadetect/run_metadetect.py [hunk 1]
```diff
@@ -69,10 +70,15 @@ def _post_process_results(*, outputs, obj_data, image_info):
 def _do_metadetect(config, mbobs, seed, i, preprocessing_function):
     _t0 = time.time()
     rng = np.random.RandomState(seed=seed)
-    if preprocessing_function is not None:
-        logger.debug("preprocessing multiband obslist %d", i)
-        mbobs = preprocessing_function(mbobs=mbobs, rng=rng)
-    res = do_metadetect(config, mbobs, rng)
+
+    n_obs = sum(len(obsl) for obsl in mbobs)
```
Comment (esheldon) [inline pizza_cutter/metadetect/run_metadetect.py]: we need all to be > 0, we don't want to do fits without some band

File: bin/run-metadetect-on-slices [hunk 2]
```diff
@@ -39,40 +40,66 @@ def _make_output_filename(meds_fname, part):
...
     help=('the part of the file to process in the '
           'form `--part="<part>:<n_parts>"` - '
           'Note that parts are one-indexed.'))
+@click.option(
+    '--range', 'meds_range', default=None, type=str,
+    help=(
+        'the range of entries in the MEDS file to process - '
+        'Note that this option is mutually exclusive with `--part`. '
+        'Futher, it is specified using normal python slicing conventions, '
+        'so that `--range="0:10"` is the first 10 entries in the file, etc.'))
```
Comment (beckermr) [inline bin/run-metadetect-on-slices]: @esheldon I added a range option which works like python slicing.

File: pizza_cutter/metadetect/run_metadetect.py (hunk 1 above)
Comment (esheldon) [reply pizza_cutter/metadetect/run_metadetect.py]: see my solution in the other

Comment (beckermr) [reply pizza_cutter/metadetect/run_metadetect.py]: yup - this is why yours is better!

File: bin/run-metadetect-on-slices (hunk 2 above)
Comment (esheldon) [reply bin/run-metadetect-on-slices]: works great, thanks

**Pull request merged**